### ISSUE TITLE
Incompatible flags between versions 3.7.2 -- 3.5.0

### ISSUE THREAD

**Issue by luchillo17** [palantir/tslint#1111]
### Bug Report
- **TSLint version**: `3.7.2--3.5.0`
- **TypeScript version**: `1.8.9`
- **Running TSLint via**: Node.js API -> webpack
### Issue

The tslint.json from a submodule references tslint-3.5.0 while project has tslint-3.7.2 which makes for incompatible flags that breaks the transpiling process, the flag is called different in both versions:
V 3.5.0: `no-trailing-comma`: boolean
             `sort-object-literal-keys`: boolean
V 3.7.2: `trailing-comma`: {}
             `object-literal-sort-keys`: boolean
#### Actual behavior

``` sh
ERROR in ./~/ng2-charts/ng2-charts.ts
Module build failed: Error: Could not find the following rules specified in the configuration:
no-trailing-comma
sort-object-literal-keys
    at Object.loadRules (/Users/aplicacionesosgroup/Documents/OSGroup/tao_app/web/node_modules/tslint/lib/ruleLoader.js:29:15)
    at Linter.lint (/Users/aplicacionesosgroup/Documents/OSGroup/tao_app/web/node_modules/tslint/lib/tslint.js:26:44)
    at Object.lint (/Users/aplicacionesosgroup/Documents/OSGroup/tao_app/web/node_modules/tslint-loader/index.js:63:23)
    at Object.module.exports (/Users/aplicacionesosgroup/Documents/OSGroup/tao_app/web/node_modules/tslint-loader/index.js:151:8)
 @ ./app/pages/dashboard/dashboard.ts 14:19-51
```
#### Expected behavior

No errors

**Comment by adidahiya** [comment]
Not exactly sure what the bug is here. We started reporting errors in [v3.4](https://github.com/palantir/tslint/releases/tag/3.4.0-dev.1) if tslint can't find a rule matching each of the configured rule names in tslint.json.

**Comment by Luchillo** [comment]
I'm using webpack with `tslint-loader` and that error about the deprecated rule names is stopping the compilation of my app, it shouldn't since a linter should only complains about bad format of code, right?

**Comment by bruno-serfe** [comment]
Same issue here:

/~/ng2-select/ng2-select.ts
Module build failed: Error: 
            Could not find implementations for the following rules specified in the configuration:
            no-trailing-comma
sort-object-literal-keys
            Try upgrading TSLint and/or ensuring that you have all necessary custom rules installed.
            If TSLint was recently upgraded, you may have old rules configured which need to be cleaned up.

**Comment by JKillian** [comment]
Is there any way to get the old version of TSLint out of your build? Ideally it'd be nicer to just be using one, up to date, version of TSLint

**Comment by johnwiseheart** [comment]
TSLint is up to 5.x now, so going to close this as hopefully nobody is still using 3.x
